rero_ils/modules/utils.py

```diff
@@ -1192,3 +1194,42 @@ def draw_row_content(row, pad=' '):
         ])+"\n"
 
     return table_header() + table_rows() + table_footer()
+
+
+def password_validator(pw, length=8, special_char=False):
+    """Validate the password.
+
+    :param pw: The password to validate.
+    :param min_length: Minimum password size.
+    :return True if the password is valid.
+    """
+    return bool(
+        len(pw) >= length
+        and len(set(string.ascii_lowercase).intersection(pw))
+        and len(set(string.ascii_uppercase).intersection(pw))
+        and len(set(string.digits).intersection(pw))
+        and (not special_char or len(set(string.punctuation).intersection(pw)))
+    )
+
+
+def password_generator(length=8, special_char=False):
+    """Generate a password.
+
+    :param min_length: Minimum password size.
+    :return the generated password.
+    """
+    min_length = 4 if special_char else 3
+    if (length < min_length):
+        raise Exception(f'Minimal size ${min_length}')
```

Comment:
dollar sign before brace must be removed
=> `raise Exception(f'Minimal size {min_length}')`